packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -0,0 +1,283 @@
+   * continue to run and remain healthy during a deployment.
+   *
+   * @default 50
+   */
+  minimumHealthyPercent?: number;
+
+  /**
+   * Time after startup to ignore unhealthy load balancer checks.
+   *
+   * @default ??? FIXME
+   */
+  healthCheckGracePeriodSeconds?: number;
+
+  /**
+   * Fargate platform version to run this service on
+   *
+   * Unless you have specific compatibility requirements, you don't need to
+   * specify this.
+   *
+   * @default Latest
+   */
+  platformVersion?: FargatePlatformVersion;
+}
+
+/**
+ * Base class for Ecs and Fargate services
+ */
+export abstract class BaseService extends cdk.Construct
+  implements elbv2.IApplicationLoadBalancerTarget, elbv2.INetworkLoadBalancerTarget, cdk.IDependable {
+
+  /**
+   * CloudFormation resources generated by this service
+   */
+  public readonly dependencyElements: cdk.IDependable[];
+
+  /**
+   * Manage allowed network traffic for this construct
+   */
+  public abstract readonly connections: ec2.Connections;
+
+  /**
+   * ARN of this service
+   */
+  public readonly serviceArn: string;
+
+  /**
+   * Name of this service
+   */
+  public readonly serviceName: string;
+
+  /**
+   * Name of this service's cluster
+   */
+  public readonly clusterName: string;
+
+  protected loadBalancers = new Array<cloudformation.ServiceResource.LoadBalancerProperty>();
+  protected networkConfiguration?: cloudformation.ServiceResource.NetworkConfigurationProperty;
+  protected readonly abstract taskDef: BaseTaskDefinition;
+  protected _securityGroup?: ec2.SecurityGroupRef;
+  private readonly resource: cloudformation.ServiceResource;
+  private scalableTaskCount?: ScalableTaskCount;
+
+  constructor(parent: cdk.Construct, name: string, props: BaseServiceProps, additionalProps: any, clusterName: string) {
+    super(parent, name);
+
+    this.resource = new cloudformation.ServiceResource(this, "Service", {
+      desiredCount: props.desiredCount || 1,
+      serviceName: props.serviceName,
+      loadBalancers: new cdk.Token(() => this.loadBalancers),
+      deploymentConfiguration: {
+        maximumPercent: props.maximumPercent || 200,
+        minimumHealthyPercent: props.minimumHealthyPercent || 50
+      },
+      /* role: never specified, supplanted by Service Linked Role */
+      networkConfiguration: new cdk.Token(() => this.networkConfiguration),
+      platformVersion: props.platformVersion,
+      ...additionalProps
+    });
+    this.serviceArn = this.resource.serviceArn;
+    this.serviceName = this.resource.serviceName;
+    this.dependencyElements = [this.resource];
+    this.clusterName = clusterName;
+  }
+
+  /**
+   * Called when the service is attached to an ALB
+   *
+   * Don't call this function directly. Instead, call listener.addTarget()
+   * to add this service to a load balancer.
+   */
+  public attachToApplicationTargetGroup(targetGroup: elbv2.ApplicationTargetGroup): elbv2.LoadBalancerTargetProps {
```

Comment:
Should we rename this to `asApplicationTargetGroupAttachment` in line with out conventions?

Reply:
Maybe, but I kinda hate that name.

Since we're  thinking about it, what about `bind`? That does something mostly similar yet again.

Reply:
what do you mean "bind"?

Reply:
https://github.com/awslabs/aws-cdk/issues/1068